yt/frontends/gadget/data_structures.py

```diff
@@ -178,8 +180,18 @@ def __init__(self, ds, io, filename, file_id, range=None):
         with header.open() as f:
             self._file_size = f.seek(0, os.SEEK_END)
 
+        default_field_type = np.dtype(io._endian + ds._header.float_type)
+        self._field_dtype = defaultdict(lambda: default_field_type)
+        self._field_dtype["ParticleIDs"] = np.dtype(io._endian + ds._id_dtype)
+
         super().__init__(ds, io, filename, file_id, range)
 
+    @contextlib.contextmanager
+    def open_handle(self):
+        handle = open(self.filename)
+        yield handle
+        handle.close()
```

Comment:
This seems like the correct way to write a @contextmanager
```suggestion
        handle = open(self.filename)
        try:
            yield handle
        finally:
            handle.close()
```

Reply:
yup, good call, I just copied over this bit and didn't think about it enough. What about just changing this block to follow the others:

```python
        with open(self.filename) as new_handle:
               yield new_handle
```

That internal `open` context should handle the closing, ya?

Reply:
yes this should be equivalent

Reply:
updated in the latest commit